### ISSUE TITLE
OrientGraphFactory.getNoTx keeps warning on IO errors

### ISSUE THREAD

**Issue by fabriziofortino** [orientechnologies/orientdb#5676]
The application uses OrientDB 2.1.8. I setup the connection pool with:

```java
private OrientGraphFactory graphFactory = new OrientGraphFactory("remote:localhost/mydb", "admin", "admin").setupPool(1, 10)
```

Sometimes I request TX connections (```graphFactory.getTx()```) other times I need NOTX connections (```graphFactory.getNoTx()```). 

When the application starts everything works as expected. If I don't use the application for a while (let's say a couple of hours) every time I get a NOTX connection I see the following warnings:

```
2016-02-02 23:19:06.312  WARN 99988 --- [io-8080-exec-10] c.o.orient.client.remote.OStorageRemote  : Caught I/O errors from Not connected (local socket=?), trying to reconnect (error: java.io.IOException: Channel is closed)
2016-02-02 23:19:06.315  WARN 99988 --- [io-8080-exec-10] c.o.orient.client.remote.OStorageRemote  : Connection re-acquired transparently after 3ms and 1 retries to server '127.0.0.1:2424/mydb': no errors will be thrown at application level
```

With TX connection I see the same warning only the first time a stale connection from the pool it's used. It seems that stale TX connections are correctly replaced but the same doesn't work if the connection is NOTX.

**Comment by lvca** [comment]
Do you have any proxy in the middle that could drop the connection after a while (timeout) ?

**Comment by fabriziofortino** [comment]
There are no proxies or other stuff that could drop the connection.

Is it correct to have TX and NOTX connections in the same pool? Should I create separate pools?

**Comment by lvca** [comment]
Do you have super long transaction could cause a timeout? Or maybe many threads with a small network pool?

**Comment by fabriziofortino** [comment]
I have changed my code. I now have two connection pools: one for TX connections, the other for NOTX.

The application now works as expected without hogging the logs with IO warnings.
When a stale connection is picked from the pool it is correctly replaced with a 'good one' (I basically see one warning for each stale connection).

It seems that OrientGraphFactory is not able to handle properly a pool with both TX and NOTX connections. I think this is reasonable. It would be good to throw an Exception (or a log warning) when an application requests both connection types from a single pool.

What do you think? If you are OK with this I could have a look at the code and create a PR.

**Comment by lvca** [comment]
Hi Fabrizio,
makes sense. If you could do a PR would be amazing!

**Comment by natemurthy** [comment]
I see very similar warnings in an application I have that uses `factory.getTx`:

```
2016-06-29 02:19:08,831 WARN  OStorageRemote - Caught I/O errors from /127.0.0.1:2424 (local socket=/127.0.0.1:48306), trying to reconnect (error: java.io.EOFException)
2016-06-29 02:19:08,832 WARN  OStorageRemote - Connection re-acquired transparently after 1ms and 1 retries to server '127.0.0.1:2424/grid': no errors will be thrown at application level
2016-06-29 02:19:09,264 WARN  OStorageRemote - Caught I/O errors from /127.0.0.1:2424 (local socket=/127.0.0.1:48314), trying to reconnect (error: java.io.EOFException)
2016-06-29 02:19:09,265 WARN  OStorageRemote - Connection re-acquired transparently after 1ms and 1 retries to server '127.0.0.1:2424/grid': no errors will be thrown at application level
2016-06-29 02:19:37,952 WARN  OStorageRemote - Caught I/O errors from Not connected (local socket=?), trying to reconnect (error: java.io.IOException: Channel is closed)
2016-06-29 02:19:37,953 WARN  OStorageRemote - Connection re-acquired transparently after 1ms and 1 retries to server '127.0.0.1:2424/grid': no errors will be thrown at application level
2016-06-29 02:19:41,339 WARN  OStorageRemote - Caught I/O errors from /127.0.0.1:2424 (local socket=/127.0.0.1:48397), trying to reconnect (error: java.io.EOFException)
2016-06-29 02:19:41,341 WARN  OStorageRemote - Connection re-acquired transparently after 2ms and 1 retries to server '127.0.0.1:2424/grid': no errors will be thrown at application level
2016-06-29 02:20:36,697 WARN  OStorageRemote - Caught I/O errors from Not connected (local socket=?), trying to reconnect (error: java.io.IOException: Channel is closed)
2016-06-29 02:20:36,698 WARN  OStorageRemote - Connection re-acquired transparently after 1ms and 1 retries to server '127.0.0.1:2424/grid': no errors will be thrown at application level
2016-06-29 02:21:37,939 WARN  OStorageRemote - Caught I/O errors from Not connected (local socket=?), trying to reconnect (error: java.io.IOException: Channel is closed)
2016-06-29 02:21:37,941 WARN  OStorageRemote - Connection re-acquired transparently after 1ms and 1 retries to server '127.0.0.1:2424/grid': no errors will be thrown at application level
2016-06-29 02:22:37,653 WARN  OStorageRemote - Caught I/O errors from Not connected (local socket=?), trying to reconnect (error: java.io.IOException: Channel is closed)
2016-06-29 02:22:37,655 WARN  OStorageRemote - Connection re-acquired transparently after 2ms and 1 retries to server '127.0.0.1:2424/grid': no errors will be thrown at application level
``

This flood of warning logs on our client typically happens when we restart the OrientDB server, and then steps when we read/write again to the database once it's back up.

**Comment by tglman** [comment]
hi @natemurthy,

It's normal that we will se a lot of this logging when a server is restarted, it should fade away after a bit of usage, we did some clean up of the logs in recent versions so as usual keep updated with the last hotfix.

**Comment by fabriziofortino** [comment]
This has been fixed in 2.2.7

Related to #6468
